main.py

```diff
@@ -433,9 +455,45 @@ def gcs_to_pubsub(cloud_event: CloudEvent):
 
     # Skip processing if filename starts with "temp"
     if "/temp_" in blob.name:
-        logging.info("Blob name contains '/temp_', skipping processing")
+        logging.debug("Blob name contains '/temp_', skipping processing")
         return
 
+    # Wait until the file stops being written to
+    initial_size = blob.size
+    retries = 0
+    max_retries = 4  # Maximum number of retries
+    while True:
+        time.sleep(30)
+        try:
+            blob.reload()  # Refresh the blob properties
+        except google_exceptions.NotFound:
+            logging.warning(
+                f"The specified blob {blob.name} does not exist in the bucket, retrying after short delay..."
+            )
+            time.sleep(30)  # Wait for 30 seconds before retrying
+            try:
+                blob.reload()  # Retry
+            except google_exceptions.NotFound:
+                logging.error(
+                    f"Retry failed, the specified blob {blob.name} still does not exist in the bucket"
+                )
+                raise  # Raise exception after failed retry
+
+        logging.debug(f"Initial size: {initial_size}, Blob size: {blob.size}")
+
+        if blob.size == initial_size:
+            logging.info("Blob size didn't change in the window proceeding")
+            break
+        elif retries >= max_retries:
+            logging.warning(
+                "Blob is still being written to after max retries. Raising an exception to let GCP handle a new retry"
+            )
+            raise
+        else:
+            logging.info("Blob size changed so we're waiting")
+            initial_size = blob.size
+            retries += 1
```

Comment:
counting retries on a else block might cause infinite loops. leave a comment on what this loop is doing